### DISCUSSION TITLE
on update start function ? ArduinoOTA.available() ?

### DISCUSSION THREAD

**Discussion by Whateverman2012** [JAndrassy/ArduinoOTA#113] [Q&A] [answered]
Hi,

 I saw onStart function in the esp8266 version of ArduinoOTA.
 Is there a way to execute code before an update with a Nano IOT 33 ?
 Something like ArduinoOTA.available() ?

 Regards.

**Comment by JAndrassy** [comment, marked as answer]
there is `void beforeApply(void (*fn)(void))`
it is invoked just before reset (at that moment the update is already stored in flash or SD)

**Reply by Whateverman2012** [reply]
perfect !
thank you.
